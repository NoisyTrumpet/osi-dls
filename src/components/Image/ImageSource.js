import React from 'react';
import PropTypes from 'prop-types';

const ImageSource = ({ srcSet, dataSrcSet, media }) => (
  <source srcSet={srcSet} data-srcset={dataSrcSet} media={media} />
);

ImageSource.defaultProps = {};

ImageSource.propTypes = {
  media: PropTypes.string,
  srcSet: PropTypes.string,
  dataSrcSet: PropTypes.string,
};

export default ImageSource;
