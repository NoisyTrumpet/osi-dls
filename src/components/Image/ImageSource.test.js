import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ImageSource from './ImageSource';

describe('<ImageSource />', () => {
  test('Should render', () => {
    const wrapper = shallow(<ImageSource />);
    expect(wrapper.getElement()).toBeTruthy();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('Should render with attributes', () => {
    const wrapper = shallow(<ImageSource srcSet="test1.jpg" dataSrcSet="data1" media="(min-width: 800px)" />);
    expect(wrapper.prop('data-srcset')).toEqual('data1');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
