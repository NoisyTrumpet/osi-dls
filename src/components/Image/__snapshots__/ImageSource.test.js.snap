// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageSource /> Should render 1`] = `<source />`;

exports[`<ImageSource /> Should render with attributes 1`] = `
<source
  data-srcset="data1"
  media="(min-width: 800px)"
  srcSet="test1.jpg"
/>
`;
